src/pages/learn/serving-over-http.mdx

```diff
@@ -34,37 +77,63 @@ http://myapi/graphql?query={me{name}}
+
+When servers that do support the `GET` method for query operations, a client may be encouraged to leverage this option to facilitate HTTP caching or edge caching in a content delivery network (CDN). However, because GraphQL documents strings may be quite long for complex operations, the query parameters may exceed the limits that browsers and CDNs impose on URL lengths.
+
+In these cases, it would be necessary to use something like _trusted documents_ (sometimes referred to as _persisted queries_) or _automatic persisted queries_, where the server stores hashes of known GraphQL documents and the clients can send a hash as the `query` parameter value in a `GET` request instead of the full source text of the query operation.
 
-`operationName` and `variables` are optional fields. `operationName` is only required if multiple operations are present in the query.
+## Response format
 
-## Response
+### Headers
 
-Regardless of the method by which the query and variables were sent, the response should be returned in the body of the request in JSON format. As mentioned in the spec, a query might result in some data and some errors, and those should be returned in a JSON object of the form:
+The response type should be `application/graphql-response+json` but can be `application/json` to support legacy clients. The server will indicate the media type in the `Content-type` header of the response. It should also indicate the encoding, otherwise `utf-8` will be assumed.
+
+### Body
+
+Regardless of the HTTP method used to send the query and variables, the response should be returned in the body of the request in JSON format. As mentioned in the GraphQL specification, a query might result in some data and some errors, and those should be returned in a JSON object of the form:
 
 ```json
 {
   "data": { ... },
-  "errors": [ ... ]
+  "errors": [ ... ],
+  "extensions": { ... }
 }
 ```
 
-If there were no errors returned, the `"errors"` field should not be present on the response. If no data is returned, [according to the GraphQL spec](https://spec.graphql.org/October2021/#sec-Data), the `"data"` field should only be included if no errors occurred during execution.
+If no errors were returned, the `errors` field should not be present in the response. If no data is returned, [according to the GraphQL spec](https://spec.graphql.org/October2021/#sec-Data), the `data` field should only be included if no errors occurred before execution. The `extensions` field is optional and information provided here will be at the discretion of the GraphQL implementation.
+
+You can read more about GraphQL spec-compliant response formats on the [Response page](/learn/response/).
+
+### Status Codes
+
+If a response contains a `data` key and its value is not `null`, then the server should respond with a `2xx` status code for either the `application/graphql-response+json` or `application/json` media types.
```

Comment:
```suggestion
If a response contains a `data` key and its value is not `null`, then the server should respond with a `2xx` status code for either the `application/graphql-response+json` or `application/json` media types. This is the case even if the response includes errors, since HTTP does not yet have a status code representing "partial success."
```